internal/jobobject/jobobject.go

```diff
@@ -436,3 +453,78 @@ func (job *JobObject) QueryStorageStats() (*winapi.JOBOBJECT_BASIC_AND_IO_ACCOUN
 	}
 	return &info, nil
 }
+
+// ApplyFileBinding makes a file binding using the Bind Filter from target to root. If the job has not been upgraded to a silo
+// this call will fail. The binding is only applied and visible for processes running in the job, any processes on the host or in another job
+// will not be able to see the binding.
+func (job *JobObject) ApplyFileBinding(root, target string, merged bool) error {
+	job.handleLock.RLock()
```

Comment:
RLock is interesting. What is the point here? If you only have a read lock it wouldnt prevent calling this twice right? And there is no state that shows its already been applied. So what is the point really?

Reply:
I forgot to check if the handle is valid at the beginning like the other methods.. (job.Handle == 0 and bail, although we should probably set this to `windows.InvalidHandle` as that seems like a better sentinel value to use thinking back). The idea is anything that modifies and invalidates the handle is behind `Lock()` while others that are just making use of the handle are RLocked